src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -306,11 +308,33 @@ def _set_rsa_parameters(dest, src):
     return client.import_key(
         vault_base_url, key_name, key_obj, destination == 'hsm', key_attrs, tags)
 
-# pylint: disable=unused-argument
-def download_secret(client, vault_base_url, secret_name, file_path, file_encoding='utf8',
-                    secret_version='', decode_binary=None):
+def download_secret(client, vault_base_url, secret_name, file_path, encoding=None,
+                    secret_version=''):
+    """ Download a secret from a KeyVault. """
+    if os.path.isfile(file_path):
```

Comment:
Consider checking for a directory as well as you can't write to a file_path if it's actually a directory.
```
if os.path.isfile(file_path) or os.path.isdir(file_path):
...
```
https://docs.python.org/2/library/os.path.html#os.path.isdir

Then, the error message would have "File or directory already exists".

Just showing you can't have a file and directory with the same name.
```
Dereks-MacBook-Pro-2:atest debekoe$ ls -la
total 0
drwxr-xr-x   2 debekoe  staff    68 Nov 11 13:42 .
drwxr-xr-x+ 63 debekoe  staff  2142 Nov 11 13:42 ..
Dereks-MacBook-Pro-2:atest debekoe$ mkdir a_directory
Dereks-MacBook-Pro-2:atest debekoe$ ls -la
total 0
drwxr-xr-x   3 debekoe  staff   102 Nov 11 13:42 .
drwxr-xr-x+ 63 debekoe  staff  2142 Nov 11 13:42 ..
drwxr-xr-x   2 debekoe  staff    68 Nov 11 13:42 a_directory
Dereks-MacBook-Pro-2:atest debekoe$ echo 'helloworld' > a_directory
-bash: a_directory: Is a directory
Dereks-MacBook-Pro-2:atest debekoe$ echo $?
1
Dereks-MacBook-Pro-2:atest debekoe$ 
```